### PR TITLE
DOCS-1284: Address dead links

### PR DESCRIPTION
**Pull request by girishshankaran**
Fixes https://tigera.atlassian.net/browse/DOCS-1284

A few URLs are corrected in the source files, and a few added to the skip list.

### CODE REVIEW

File: __tests__/crawler.test.js [hunk 1]
```diff
@@ -31,6 +31,21 @@ test("Test links to check if they're all reachable", async () => {
...
+    `https://docs.tigera.io/calico/charts`,
+    `https://sysctl-explorer.net/net/ipv4/fib_multipath_hash_policy`,
+    `https://github.com/projectcalico/calico/releases/download/master/ocp.tgz`,
+    `https://docs.tigera.io/master/reference/installation/api#operator.tigera.io%2fv1`,
+    `https://downloads.tigera.io/ee/master/download/binaries/master/calicoctl`,
+    `https://projectcalico.docs.tigera.io/manifests/calico-vxlan.yaml`,
+    ///^https?:\/\/localhost[\w\-.~:/?#[\]@!$&'()*+,;=]*$/,
+    /^https?:\/\/csrc\.nist\.gov\/projects\/cryptographic-module-validation-program\/certificate\/\d+$/,
+    /^https:\/\/installer\.calicocloud\.io:[0-9]{3,4}$/,
+    /^https:\/\/github\.com\/[\w-]+\/[\w-]+\/releases\/download\/[\w-]+\/install-calico-windows\.ps1$/,
```
Comment (doucol) [inline __tests__/crawler.test.js]: Should some of these be fixed in the source docs? We should always try to fix in the source docs before resorting to the skip list. Please review each of these with @ctauchen  - thank you.

**Review by doucol — CHANGES_REQUESTED** [review]

File: __tests__/crawler.test.js (hunk 1 above)
Comment (girishshankaran) [reply __tests__/crawler.test.js]: Sure Doug.

Comment (ctauchen) [reply __tests__/crawler.test.js]: Yeah, I'd say a few of those just need fixing at source:
* http://heartbleed.com has been dead quite a while now. Can we find an alternative or remove it?
* https://sysctl-explorer.net/net/ipv4/fib_multipath_hash_policy/ (It just needs the trailing slash! `/`)
* https://projectcalico.docs.tigera.io/manifests/calico-vxlan.yaml --> That definitely needs changing to a new URL with {{manifestsUrl}}. But I'm having trouble finding where this is appearing in the docs. Have you seen it?
* https://docs.tigera.io/master/reference/installation/api#operator.tigera.io%2fv1 Can definitely be changed.
* The .ps1 URL seems to be for all versions? I would expect to only be for /master/ for OS. 

One category of broken links is a number of 'master' resources, which are known not to be present. There's more involved in sorting these out. Last I heard, there was no immediate plan to maintain these. The idea is that we don't want people actually building stuff from /master/?

@girishshankaran I'll write a ticket to deal with these /master/ resources separately. For now, let's have them in the skip list.

Comment (girishshankaran) [reply __tests__/crawler.test.js]: - http://heartbleed.cm: Found an [alternative link](https://owasp.org/www-community/vulnerabilities/Heartbleed_Bug).
- https://sysctl-explorer.net/net/ipv4/fib_multipath_hash_policy/: The trailing slash is there in the source. Check with @doucol why it is reported as dead. Does the crawler ignore trailing slashes in a URL?
- https://docs.tigera.io/master/reference/installation/api#operator.tigera.io%2fv1: Changed in the source
- The specific .ps1 URL is added to the skip list
- https://projectcalico.docs.tigera.io/manifests/calico-vxlan.yaml: Changed in the source. @ctauchen Please verify the fix.

Comment (girishshankaran) [reply __tests__/crawler.test.js]: Let me run the test :)

Comment (ctauchen) [reply __tests__/crawler.test.js]: Getting close! My results from the test:
image

Comment (girishshankaran) [reply __tests__/crawler.test.js]: Added these links in the skip list:
```
http://localhost:4242/calico/3.24/network-policy/istio/files/30-policy.yaml
https://downloads.tigera.io/ee/master/download/binaries/master/calicoq
https://en.wikipedia.org/wiki/Autonomous_System_(Internet)%3E
 https://installer.calicocloud.io/manifests/v3.15.1-8/manifests
```

**Review by ctauchen — DISMISSED** [review]
LGTM!!! Thanks!

File: __tests__/crawler.test.js (hunk 1 above)
Comment (doucol) [reply __tests__/crawler.test.js]: I am curious about this one:
http://localhost:4242/calico/3.24/network-policy/istio/files/30-policy.yaml

Why is this not something we are fixing in the source docs?

Comment (doucol) [reply __tests__/crawler.test.js]: hint:  if you do a search for "(files/30-policy.yaml" - you will find the offending reference. It is, in fact, a bug in our source docs.

Comment (ctauchen) [reply __tests__/crawler.test.js]: @girishshankaran can you take a look at this?

Comment (girishshankaran) [reply __tests__/crawler.test.js]: Yes Chris. I fixed the issue this morning. The issue seems to be with a missing slash (/) in the file path. @doucol Please correct me if I am wrong. I removed the link from the skip list as well. We need to run a test to make sure that the issue is resolved.

File: calico_versioned_docs/version-3.25/operations/ebpf/install.mdx [hunk 2]
```diff
@@ -302,7 +302,7 @@ For example:
 
 ```yaml
  This section includes base Calico Enterprise installation configuration.
- For more information, see: https://docs.tigera.io/calico-enterprise/latest/reference/installation/api/#operator.tigera.io/v1
+ For more information, see: https://docs.tigera.io/calico/3.25/reference/installation/api/#operator.tigera.io/v1
```
Comment (doucol) [inline calico_versioned_docs/version-3.25/operations/ebpf/install.mdx]: Why are we changing from 'latest' to 3.25 here?  @ctauchen

**Review by doucol — CHANGES_REQUESTED** [review]

File: calico_versioned_docs/version-3.24/network-policy/istio/enforce-policy-istio.mdx [hunk 3]
```diff
@@ -144,7 +144,7 @@ calicoctl create -f 30-policy.yaml
 :::note
 
 You can also
-[view the manifest in your browser](files/30-policy.yaml).
+[view the manifest in your browser](/files/30-policy.yaml).
```
Comment (doucol) [inline calico_versioned_docs/version-3.24/network-policy/istio/enforce-policy-istio.mdx:147]: @ctauchen @girishshankaran yes, this is the forward slash that was missing. Without the forward slash, it's a relative path (to this mdx file), which, of course, does not exist.

File: __tests__/crawler.test.js (hunk 1 above)
Comment (doucol) [reply __tests__/crawler.test.js]: > Yes Chris. I fixed the issue this morning. The issue seems to be with a missing slash (/) in the file path. @doucol Please correct me if I am wrong. I removed the link from the skip list as well. We need to run a test to make sure that the issue is resolved.

That is correct. It was missing a leading forward slash to make the path absolute, not relative to the mdx file.

Comment (girishshankaran) [reply __tests__/crawler.test.js]: Strangely, the link was working fine in the 3.24 website.

**Comment by doucol** [comment]
@ctauchen @girishshankaran this needs to get merged today don't you think?

File: calico_versioned_docs/version-3.25/operations/ebpf/install.mdx (hunk 2 above)
Comment (ctauchen) [reply calico_versioned_docs/version-3.25/operations/ebpf/install.mdx]: I'm not sure. @girishshankaran ? I suspect the intent here was to keep someone using this YAML file within the same version. But I know in other YAML files, which tend to be shared from one version to the next, we just always point to /latest/.

**Comment by ctauchen** [comment]
> @ctauchen @girishshankaran this needs to get merged today don't you think?

Yes!

**Comment by girishshankaran** [comment]
Closing this PR as there is another [PR](https://github.com/tigera/docs/pull/471) that addresses these changes.

**Pull request closed without merging**